components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -785,21 +804,18 @@ public static String getUserConsentURL(OAuth2Parameters params, String loggedInU
                 if (entry != null) {
                     user = entry.getLoggedInUser();
                 }
-                setConsentRequiredScopesToOAuthParams(user, params);
-                Set<String> consentRequiredScopesSet = params.getConsentRequiredScopes();
-                String consentRequiredScopes = StringUtils.EMPTY;
-                if (CollectionUtils.isNotEmpty(consentRequiredScopesSet)) {
-                    consentRequiredScopes = String.join(" ", consentRequiredScopesSet).trim();
-                }
+                String consentRequiredScopes = getConsentRequiredScopes(params, user);
 
                 consentPage = consentPage + "&" + OAuthConstants.OAuth20Params.SCOPE + "=" + URLEncoder.encode
                         (consentRequiredScopes, UTF_8) + "&" + OAuthConstants.SESSION_DATA_KEY_CONSENT
                         + "=" + URLEncoder.encode(sessionDataKeyConsent, UTF_8) + "&" + "&spQueryParams=" + queryString;
 
-                if (entry != null) {
+                // Append additional query params to the consent page url.
+                consentPage = FrameworkUtils.appendQueryParamsStringToUrl(consentPage, additionalQueryParams);
 
-                    consentPage = FrameworkUtils.getRedirectURLWithFilteredParams(consentPage,
-                            entry.getEndpointParams());
+                if (entry != null) {
+                    // Filter the query parameters from the consent page url.
+                    consentPage = getConsentPageURLWithFilteredParams(entry, consentPage);
```

Comment:
Change the method name

Reply:
Resolved by https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2020/commits/ba3d3ded0c372f6c6846cbb176d101e83a71c105